node-db/lesson2.md

```diff
@@ -1,22 +1,390 @@
+# Before we start
+Fork and Clone the repo - https://github.com/CodeYourFuture/express-workshop-2
 
-## 11:00 AM - Scrum session with Kram
+# Recap
 
-## 1:00 PM - Node session
+- **Package.json**: Remember how in the last lesson we added a *package.json* file and added dependencies like *express* to it.
 
-This is going to be a mixture of instructor-led parts to recap the fundamentals of Node.js, and practical pair-programming parts to get more experience with building Node.js apps.
+    - What command did we use to create a *package.json* file
+    - How did we add dependencies to it?
 
-### Node.js Fundamentals Recap
+    Today, the project comes with a package file already - this lists all the dependencies that we need to run the project (instead of adding them to the Git repo, we just communicate the fact that we need them). Now run `npm install` and your dependencies will be installed.
 
-[Link](/node-recap.md)
+- **Server.js** - this is the entry point for our application. Let's walk through the code and remember what it does.
+    - What is **Express Static Middleware** used for?
+    - What routes are defined?
+    - What **Port** does the server run on?
 
-### Practical exercises: The Status Code API and The Status Code Website.
+> Let's run the server and check it on the browser. This is what you should see on your browser
+![](assets/blog-screenshot.png)
 
-Part 1. [Build a Node status code API](https://github.com/Code-Your-Future/node-status-code-api)
+* The theme used for this website is based on Bootstrap. You can get more open source Bootsrap-based themese from [StartBootstrap](https://startbootstrap.com/template-categories/all/)
 
-Part 2. [Build a Node status code website](https://github.com/Code-Your-Future/node-status-code-website)
-0Looking
+> **Exercise**: Notice how the link to **Contact me** doesn't return a page - Let's add an enpoint to return a simple string **this is a contact page**.
+
+# Templating
+> **Exercise**: At Let's change the text for the link for **Contact** from **Contact me** to **Contact Information**.
+
+    - How many files do you have to change? Wouldn't be nice if we could change one file and that reflects in all pages.
+
+At the moment, we're just serving static HTML files from the *public* folder. **NodeJS** and **ExpressJS** allows us to build dynamic web pages.
+
+> A server-side dynamic web page is a web page whose construction is controlled by an application server processing server-side scripts. In server-side scripting, parameters determine how the assembly of every new web page proceeds, including the setting up of more client-side processing.
+
+## Template Engines
+A template engine enables you to use static template files in your application. At **runtime**, the template engine replaces variables in a template file with actual values, and transforms the template into an HTML file sent to the client. This approach makes it easier to design an HTML page, as it reduces duplicate code (DRY).
```

Comment:
Worth explaining what `static template files` are? They've just been exposed to HTML files up to now.